test/conformance/certificate/utils.go

```diff
@@ -74,23 +78,42 @@ func CreateCertificate(t *testing.T, clients *test.Clients, dnsNames []string) (
 	}
 }
 
-// WaitForSecret polls the status of the named Secret until it exists or the timeout is exceeded
-func WaitForSecret(client *test.Clients, name string, desc string) error {
-	span := logging.GetEmitableSpan(context.Background(), fmt.Sprintf("waitForSecret/%s/%s", name, desc))
+// WaitForCertificateSecret polls the status of the Secret for the provided Certificate
+// until it exists or the timeout is exceeded. It then validates its contents
+func WaitForCertificateSecret(client *test.Clients, cert *v1alpha1.Certificate, desc string) error {
+	span := logging.GetEmitableSpan(context.Background(), fmt.Sprintf("WaitForCertificateSecret/%s/%s", cert.Spec.SecretName, desc))
 	defer span.End()
 
 	waitErr := wait.PollImmediate(test.PollInterval, test.PollTimeout, func() (bool, error) {
-		_, err := client.KubeClient.Kube.CoreV1().Secrets(test.ServingNamespace).Get(name, metav1.GetOptions{})
+		secret, err := client.KubeClient.Kube.CoreV1().Secrets(test.ServingNamespace).Get(cert.Spec.SecretName, metav1.GetOptions{})
 
 		if apierrs.IsNotFound(err) {
 			return false, nil
+		} else if err != nil {
+			return true, err
+		}
+
+		block, _ := pem.Decode(secret.Data[corev1.TLSCertKey])
+		if block == nil {
+			return true, fmt.Errorf("Failed to decode PEM data")
```

Comment:
No formatting, but don't ignore error.

Reply:
I'll change the `fmt` call. `pem.Decode` doesn't return an error though. It returns the first PEM-formatted block and the rest of the input. In the case of invalid PEM data it returns `nil` and the entire input.

Reply:
```suggestion
			return true, errors.New("failed to decode PEM data")
```